linera-views/src/reentrant_collection_view.rs

```diff
@@ -326,6 +326,168 @@ where
     }
 }
 
+impl<C, W> ReentrantByteCollectionView<C, W>
+where
+    C: Context + Send + Clone + 'static,
+    ViewError: From<C::Error>,
+    W: View<C> + Send + Sync + 'static,
+{
+    /// Load multiple entries for write at once.
+    /// The entries in short_keys have to be all distincts.
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::{create_test_context, MemoryContext};
+    /// # use linera_views::reentrant_collection_view::ReentrantByteCollectionView;
+    /// # use linera_views::register_view::RegisterView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_context();
+    ///   let mut view : ReentrantByteCollectionView<_, RegisterView<_,String>> = ReentrantByteCollectionView::load(context).await.unwrap();
+    ///   {
+    ///     let mut subview = view.try_load_entry_mut(vec![0, 1]).await.unwrap();
+    ///     *subview.get_mut() = "Bonjour".to_string();
+    ///   }
+    ///   let short_keys = vec![vec![0, 1], vec![2, 3],];
+    ///   let subviews = view.try_load_entries_mut(short_keys).await.unwrap();
+    ///   let value1 = subviews[0].get();
+    ///   let value2 = subviews[1].get();
+    ///   assert_eq!(*value1, "Bonjour".to_string());
+    ///   assert_eq!(*value2, String::default());
+    /// # })
+    /// ```
+    pub async fn try_load_entries_mut(
+        &mut self,
+        short_keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<OwnedRwLockWriteGuard<W>>, ViewError> {
+        let mut selected_short_keys = Vec::new();
+        *self.hash.get_mut() = None;
+        let updates = self.updates.get_mut();
+        for short_key in short_keys.clone() {
+            match updates.entry(short_key.clone()) {
+                btree_map::Entry::Occupied(entry) => {
+                    let entry = entry.into_mut();
+                    if let Update::Removed = entry {
+                        selected_short_keys.push((short_key, true));
+                    }
+                }
+                btree_map::Entry::Vacant(_entry) => {
+                    selected_short_keys.push((short_key, self.was_cleared));
+                }
+            }
+        }
+        let mut handles = Vec::new();
+        for short_key in &selected_short_keys {
+            let short_key = short_key.0.clone();
+            let context = self.context.clone();
+            handles.push(tokio::spawn(async move {
+                let key = context.base_tag_index(KeyTag::Subview as u8, &short_key);
+                let context = context.clone_with_base_key(key);
+                W::load(context).await
+            }));
+        }
+        let response = futures::future::join_all(handles).await;
+        for (i, view) in response.into_iter().enumerate() {
+            let short_key = &selected_short_keys[i];
```

Comment:
`let (short_key, &to_be_cleared) = &selected_short_keys[i];`

Reply:
Ok for restructuration but that syntax is incorrect. It should be
`let (shortkey, to_be_cleared) = &selected_short_keys[i]`
and `to_be_cleared` will be of type `&bool`.